src/main.rs

```diff
@@ -161,6 +167,23 @@ fn check_host(target: &CanaryTarget) -> CanaryCheck {
     }
 }
 
+
+/*
+    Checks if alert would not be spam.
+    Alert would not be spam iff state changes from "okay" to "fire" or "fire" to "okay".
+    Updates HashMap last_statuses with last state seen for the target.
+*/
+fn check_not_spam(last_statuses: &mut HashMap<CanaryTarget, String>, result: &CanaryCheck) -> bool {
```

Comment:
Will rewrite to `check_spam` removing the negation.

To confirm the expected behavior, if there is no previous history (`None` case), then the alert should be sent?